scripts/build/builders/android.py

```diff
@@ -130,7 +130,7 @@ def _build(self):
         self._Execute(['mkdir', '-p', jnilibs_dir],
                       title='Prepare Native libs ' + self.identifier)
 
-        for libName in ['libSetupPayloadParser.so', 'libCHIPController.so']:
+        for libName in ['libSetupPayloadParser.so', 'libCHIPController.so', 'libc++_shared.so']:
```

Comment:
Can we please use the builtin runtime deps functionality to create this list?

Reply:
The command:

```
gn desc out/android-x64-chip_tool/ //src/setup_payload/java runtime_deps
```

seems to work well in listing required data, however I am suffering here from a design issue: the build script attempts to support `dry-run` for testing and only lists out the build commands. In this case, the build commands would depend in `gn gen` actually having run, so that path would not work for dry run.

I would prefer to do this separately (if at all ... I am  not yet clear on the solution, ideally maybe we should merge the two so files and not need libc++_shared.so at all) because of this.

Reply:
Added a TODO explaining this issue. I believe the long-term solution is likely to not need libc++_shared.so anymore by merging jni files.